pkgs/applications/graphics/drawio/default.nix

```diff
@@ -0,0 +1,94 @@
+{ stdenv, fetchurl, rpmextract, wrapGAppsHook,
+
+# Dynamic libraries
+alsaLib, atk, at-spi2-atk, at-spi2-core, cairo, dbus, cups, expat,
+gdk_pixbuf, glib, gtk3, libX11, libXScrnSaver, libXcomposite, libXcursor,
+libXdamage, libXext, libXfixes, libXi, libXrandr, libXrender, libXtst,
+libxcb, libuuid, nspr, nss, pango,
+
+# Crashes without systemd.
+systemd
+}:
+
+let
+  rpath = stdenv.lib.makeLibraryPath [
+   alsaLib
+    atk
+    at-spi2-atk
+    at-spi2-core
+    cairo
+    cups
+    dbus
+    expat
+    gdk_pixbuf
+    glib
+    gtk3
+    libX11
+    libXScrnSaver
+    libXcomposite
+    libXcursor
+    libXdamage
+    libXext
+    libXfixes
+    libXi
+    libXrandr
+    libXrender
+    libXtst
+    libxcb
+    libuuid
+    nspr
+    nss
+    pango
+    systemd
+  ];
+in stdenv.mkDerivation rec {
+  pname = "drawio";
+  version = "10.8.0";
+
+  src = fetchurl {
+    url = "https://github.com/jgraph/drawio-desktop/releases/download/v${version}/draw.io-x86_64-${version}.rpm";
+    sha256 = "0c5wymzhbp72x0yhvw7vb4akkdvj97npl9kglk79vqjbzfn5di9k";
+  };
+
+  nativeBuildInputs = [ rpmextract wrapGAppsHook ];
+
+  phases = [ "unpackPhase" "installPhase" ];
+
+  unpackPhase = "rpmextract ${src}";
+
+  installPhase = ''
+    mkdir -p $out/libexec
+    cp -r opt/draw.io $out/libexec/
+
+    # Application icon
+    mkdir -p $out/share/icons/hicolor
+    cp -r usr/share/icons/hicolor/0x0 $out/share/icons/hicolor/1024x1024
+
+    # XDG desktop item
+    cp -r usr/share/applications $out/share/applications
+
+    patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+             --set-rpath ${rpath}:$out/libexec/draw.io $out/libexec/draw.io/draw.io
+
+    wrapProgram $out/libexec/draw.io/draw.io \
```

Comment:
It would be cleaner to do

```
preFixup = ''
  gappsWrapperArgs+=(
    --prefix LD_LIBRARY_PATH : "${stdenv.cc.cc.lib}/lib"
  )
'';
```

Reply:
Done, I have also added `systemd` to the path here (since the library path wasn't isn't added through `patchelf` anymore).